java/src/jmri/jmrit/logixng/expressions/ExpressionClock.java

```diff
@@ -162,80 +186,103 @@ public String getLongDescription(Locale locale) {
     public void setup() {
         // Do nothing
     }
-    
-    /** {@inheritDoc} */
+
+    /** {@inheritDoc}
+     * The SystemClock listener creates a timer on the first call.  Subsequent calls
+     * enabled timer processing.
+     */
     @Override
     public void registerListenersForThisClass() {
         if (!_listenersAreRegistered) {
             switch (_type) {
                 case SystemClock:
-                    throw new UnsupportedOperationException("Not implemented yet");
-                    
+                    if (timerDefined) {
+                        timerRunning = true;
+                    } else {
+                        minuteTimer();
+                        timerDefined = true;
+                        timerRunning = true;
+                    }
+                    break;
+
                 case FastClock:
                     _fastClock.addPropertyChangeListener("time", this);
                     break;
-                    
+
                 default:
                     throw new UnsupportedOperationException("_type has unknown value: " + _type.name());
             }
-            
+
             _listenersAreRegistered = true;
         }
     }
-    
-    /** {@inheritDoc} */
+
+    /** {@inheritDoc}
+     * The SystemClock timer flag is set false to suspend processing of timer events.  The
+     * timer keeps running for the duration of the JMRI session.
+     */
     @Override
     public void unregisterListenersForThisClass() {
         if (_listenersAreRegistered) {
             switch (_type) {
                 case SystemClock:
-                    throw new UnsupportedOperationException("Not implemented yet");
-                    
+                    timerRunning = false;
+                    break;
+
                 case FastClock:
-                    _fastClock.removePropertyChangeListener("time", this);
+                    if (_fastClock != null) _fastClock.removePropertyChangeListener("time", this);
                     break;
-                    
+
                 default:
                     throw new UnsupportedOperationException("_type has unknown value: " + _type.name());
             }
-            
+
             _listenersAreRegistered = false;
         }
     }
-    
+
     /** {@inheritDoc} */
     @Override
     public void propertyChange(PropertyChangeEvent evt) {
         if (getTriggerOnChange()) {
             getConditionalNG().execute();
         }
     }
-    
+
     /** {@inheritDoc} */
     @Override
     public void disposeMe() {
     }
```

Comment:
I think it would be good if this method cancelled the timer. It's called when this expression is deleted.

Reply:
I tried several approaches but none of them were reliable.  With the current approach, I am not aware of any way to find the TimerTask.

Reply:
The timer for the system clock has been refactored.